### ISSUE TITLE
Iterator issue

### ISSUE THREAD

**Issue by achouba** [CosyVerif/library#34]
here is the function that I used:

iterator = "return function (yield, store)\
    for k in pairs (store.user) do\
      yield (store.user [k].position)\
    end\
  end"
  })
The result is empty.

but I can get all other values hidden,private and public!!

**Comment by achouba** [comment]
Now I can't access to private and hidden values but still cant' get position

**Comment by saucisson** [comment]
You cannot get a whole table (for instance `position`). Instead, extract the fields that you need:

````lua
    return function (yield, store)
      for k in pairs (store.user) do
        yield { username = store.user [k].username, timezone = store.user [k].position.timezone }
      end
    end
````

**Comment by saucisson** [comment]
Fixed in eb575535dbdb893e4be929466b205aea3bff4e92
